habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -30,7 +30,23 @@ class ObjectNavDatasetV1(PointNavDatasetV1):
     content_scenes_path: str = "{data_path}/content/{scene}.json.gz"
 
     def to_json(self) -> str:
+        self.goals_per_category = {}
+        for i, ep in enumerate(self.episodes):
+            goals_id = "{}_{}".format(ep.scene_id, ep.goals[0].object_id)
+            if goals_id not in self.goals_per_category:
+                self.goals_per_category[goals_id] = ep.goals
+
+            self.episodes[i].goals = goals_id
```

Comment:
Key can be recreated from episode data:
```suggestion
            self.episodes[i].goals = []
```